pkg/scheduler/util/backoff_utils_test.go

```diff
@@ -64,23 +64,33 @@ func TestBackoff(t *testing.T) {
 	}
 
 	for _, test := range tests {
-		duration := backoff.GetEntry(test.podID).getBackoff(backoff.maxDuration)
-		if duration != test.expectedDuration {
-			t.Errorf("expected: %s, got %s for %s", test.expectedDuration.String(), duration.String(), test.podID)
-		}
-		clock.t = clock.t.Add(test.advanceClock)
-		backoff.Gc()
+		t.Run(test.expectedDuration.String(), func(t *testing.T) {
+			duration := backoff.GetEntry(test.podID).getBackoff(backoff.maxDuration)
+			if duration != test.expectedDuration {
+				t.Errorf("expected: %s, got %s for %s", test.expectedDuration.String(), duration.String(), test.podID)
+			}
+
+			clock.t = clock.t.Add(test.advanceClock)
+			backoff.Gc()
+		})
 	}
+
 	fooID := ktypes.NamespacedName{Namespace: "default", Name: "foo"}
 	backoff.perPodBackoff[fooID].backoff = 60 * time.Second
-	duration := backoff.GetEntry(fooID).getBackoff(backoff.maxDuration)
-	if duration != 60*time.Second {
-		t.Errorf("expected: 60, got %s", duration.String())
-	}
-	// Verify that we split on namespaces correctly, same name, different namespace
-	fooID.Namespace = "other"
-	duration = backoff.GetEntry(fooID).getBackoff(backoff.maxDuration)
-	if duration != 1*time.Second {
-		t.Errorf("expected: 1, got %s", duration.String())
-	}
+	t.Run("Verify that we split on namespaces correctly, same name, different namespace", func(t *testing.T) {
```

Comment:
I don't think this t.Run improves the clarity of the test. Similarly, the two subtests below might make more sense as better error messages.

Reply:
agreed. i meant to come back and revert this bit, as its a bit out of scope for the add subtests on table driven units objective.